core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -194,15 +192,34 @@ public static String decodeString(String encoded) {
    * @return UTF-8 encoded string representing the namespace, suitable for use as a URL parameter
    */
   public static String encodeNamespace(Namespace ns) {
-    Preconditions.checkArgument(ns != null, "Invalid namespace: null");
-    String[] levels = ns.levels();
+    return encodeNamespace(ns, NAMESPACE_ESCAPED_SEPARATOR);
+  }
+
+  /**
+   * Returns a String representation of a namespace that is suitable for use in a URL / URI.
```

Comment:
nit: this description is too vague. It does not clarify whether URL-encoding has been applied or whether the returned string will need to go through the URL encoding process later (i.e. at HTTP request time).

Reply:
I specifically wanted to avoid changing anything around the existing code & javadoc. This is really just a copy/paste of the existing javadoc for the overloaded method